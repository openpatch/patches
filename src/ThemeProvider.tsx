import React, { ReactElement } from "react";
import { ThemeProvider as EThemeProvider } from "emotion-theming";
import { Theme, base } from "./themes";
import { GlobalCssTheme } from "./themes/utils";

interface ThemeProviderProps {
  theme?: Theme;
  children: ReactElement;
}

const ThemeProvider = ({ theme = base, children }: ThemeProviderProps) => {
  return (
    <EThemeProvider theme={theme}>
      <GlobalCssTheme theme={theme} />
      {children}
    </EThemeProvider>
  );
};

export default ThemeProvider;
