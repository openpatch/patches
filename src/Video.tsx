import styled from "@emotion/styled";
import { HTMLAttributes } from "react";
import {
  border,
  BorderProps,
  shadow,
  ShadowProps,
  space,
  SpaceProps,
} from "./system";

export type VideoProps = {
  src: string;
  loop?: boolean;
  autoPlay?: boolean;
} & BorderProps &
  SpaceProps &
  ShadowProps &
  HTMLAttributes<HTMLVideoElement>;

export const StyledVideo = styled.video<
  BorderProps & SpaceProps & ShadowProps & HTMLAttributes<HTMLVideoElement>
>(border, space, shadow);

export const Video = ({ src, loop, autoPlay, ...props }: VideoProps) => {
  return (
    <StyledVideo loop={loop} autoPlay={autoPlay} {...props}>
      <source src={src} />
    </StyledVideo>
  );
};
