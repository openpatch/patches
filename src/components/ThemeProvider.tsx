import React, { ReactElement } from "react";
import {
  ThemeProvider as SCThemeProvider,
  DefaultTheme,
} from "styled-components";
import defaultTheme from "../shared/theme";

interface ThemeProviderProps {
  theme?: DefaultTheme;
  children: ReactElement;
}

const ThemeProvider = ({
  theme = defaultTheme,
  children,
}: ThemeProviderProps) => {
  return <SCThemeProvider theme={theme}>{children}</SCThemeProvider>;
};

export default ThemeProvider;
