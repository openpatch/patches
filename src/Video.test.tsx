import React from "react";
import { render } from "@testing-library/react";
import { Video, VideoProps } from "./Video";
import { ThemeProvider } from "./ThemeProvider";

const renderWithTheme = (props?: VideoProps) =>
  render(
    <ThemeProvider>
      <Video
        {...props}
        src="https://f002.backblazeb2.com/file/openpatch-static/interactive.mp4"
      />
    </ThemeProvider>
  );

test("should render", () => {
  renderWithTheme();
});

test("should match snapshot", () => {
  const { container } = renderWithTheme();
  expect(container).toMatchSnapshot();
});
