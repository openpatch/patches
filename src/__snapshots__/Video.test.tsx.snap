// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div>
  <video
    class="emotion-0"
  >
    <source
      src="https://f002.backblazeb2.com/file/openpatch-static/interactive.mp4"
    />
  </video>
</div>
`;
